manager/orchestrator/constraintenforcer/constraint_enforcer.go

```diff
@@ -115,16 +122,27 @@ func (ce *ConstraintEnforcer) rejectNoncompliantTasks(node *api.Node) {
 		// Ensure that the task assigned to the node
 		// still satisfies the resource limits.
 		if t.Spec.Resources != nil && t.Spec.Resources.Reservations != nil {
-			if t.Spec.Resources.Reservations.MemoryBytes > availableMemoryBytes {
+			if t.Spec.Resources.Reservations.MemoryBytes > available.MemoryBytes {
 				removeTasks[t.ID] = t
 				continue
 			}
-			if t.Spec.Resources.Reservations.NanoCPUs > availableNanoCPUs {
+			if t.Spec.Resources.Reservations.NanoCPUs > available.NanoCPUs {
 				removeTasks[t.ID] = t
 				continue
 			}
-			availableMemoryBytes -= t.Spec.Resources.Reservations.MemoryBytes
-			availableNanoCPUs -= t.Spec.Resources.Reservations.NanoCPUs
+			for _, ta := range t.AssignedGenericResources {
+				// Type change or no longer available
+				if genericresource.HasResource(ta, available.Generic) {
+					removeTasks[t.ID] = t
+					break loop
+				}
+			}
+
+			available.MemoryBytes -= t.Spec.Resources.Reservations.MemoryBytes
+			available.NanoCPUs -= t.Spec.Resources.Reservations.NanoCPUs
+
+			genericresource.ClaimResources(&available.Generic,
+				&fakeStore, t.AssignedGenericResources)
```

Comment:
Can you explain why this uses `ClaimResources` instead of `RemoveNodeResources`? I notice that the scheduler uses `RemoveNodeResources` when reconciling.